Add `x-forwarded-*` headers to upstream lambda request

fluxth opened the pull request:
Lambda event from real API gateway always have these `x-forwarded-*` headers, and some library that translate lambda events for HTTP frameworks expects these.

[Mangum](https://github.com/jordaneremieff/mangum) will default to HTTPS if there's no `x-forwarded-proto` present in the request headers.

fluxth committed:
Add `x-forwarded-*` headers to upstream lambda request

mnapoli commented:
Thank you, that sounds great! Could you update the tests too?

fluxth committed:
Update tests

fluxth commented:
Tests updated 🚀

mnapoli commented on src/index.ts:
```diff
@@ -128,5 +128,3 @@ type LambdaInvokeError = {
     errorMessage: string;
 };
-
-export default app;
```
That is a breaking change right?

fluxth replied:
Ah, I didn't see usage of this default export anywhere, but I'm happy to keep this if you prefer.

fluxth replied:
Fixed in 5c1add86781dfee6d685c751d3083c158826aa56

mnapoli replied:
Thanks! Since it's exported on NPM some people might use it via the programmatic API.

fluxth committed:
Keep default export in `index.ts`

fluxth committed:
Merge remote-tracking branch 'origin/main' into x-forwarded-headers

fluxth committed:
Update IP addresses in tests because we use `app.listen`

We explicitly use bound express server to IPv4 `0.0.0.0`, therefore the
IP address we get back won't be V4MAP `::ffff:0:0/96` range anymore.
